PxGet/Functions/Invoke-PxGet.ps1

```diff
@@ -0,0 +1,133 @@
+function Invoke-PxGet
+{
+    <#
+    .SYNOPSIS
+    Invokes PxGet.
+    
+    .DESCRIPTION
+    A tool similar to nuget but for PowerShell modules. A config file in the root of a repository that specifies 
+    what modules should be installed into the PSModules directory of the repository. 
+
+    .EXAMPLE
+    Invoke-PxGet 'install'
+
+    Demonstrates how to call this function to install required PSModules.
+    #>
+
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [ValidateSet('install')]
+        [String] $Command
+    )
+ 
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+    $DebugPreference = 'Continue'
+
+    $origModulePath = $env:PSModulePath
+    $psmodulesPath = Join-Path -Path $(Get-RootDirectory) -ChildPath 'PSModules'
+    if( -not (Test-Path -Path $psmodulesPath) )
+    {
+        New-Item -Path $psmodulesPath -ItemType 'Directory' | Out-Null
+    }
+    
+    try
+    {
+        # pxget should ship with its own private copies of PackageManagement and PowerShellGet. Setting PSModulePath
+        # to pxget module's Modules directory ensures no other package modules get loaded.
+        $env:PSModulePath = Join-Path -Path $(Get-RootDirectory) -ChildPath 'PSModules' -Resolve
+        Import-Module -Name 'PackageManagement'
+        Import-Module -Name 'PowerShellGet'
+
+        $moduleNames = @()
+        $pxModules = Get-Content -Path ($(Get-RootDirectory) + '\pxget.json') | ConvertFrom-Json
+ 
+        foreach ($pxModule in $pxModules.PSModules) 

```

Comment:
Replace the `foreach` block with: `$moduleNames = $pxModules | Select-Object -ExpandProperty 'Name'` and remove the `$moduleNames` assignment on line 43.